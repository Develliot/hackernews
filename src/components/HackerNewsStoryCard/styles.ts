import styled from 'styled-components';

export const StoryCardWrapper = styled.div`
    border: solid 1px black;

    transition: all 1s ease;

    /* &:hover {
        grid-column: span 2;
        grid-row: span 2;
    } */
`;
