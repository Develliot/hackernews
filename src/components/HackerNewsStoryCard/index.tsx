import React, { FunctionComponent, memo } from 'react';

import { ErrorMessage } from 'src/components/ErrorMessage';
import { StoryCardWrapper } from './styles';

import { StoryType } from 'src/containers/HackerNewsStoryContainer';

type Props = {
    isError: boolean;
    isLoading: boolean;
    story: StoryType;
    retry: () => void;
};

export const HackerNewsStoryCard: FunctionComponent<Props> = ({
    isError,
    isLoading,
    story,
    retry,
}) => {
    return (
        <>
            {isError ? (
                <ErrorMessage
                    errorMessage='Unable to retrieve story'
                    retry={retry}
                />
            ) : isLoading ? (
                <>
                    <span>Loading story</span>
                    {/* TODO: loading state */}
                    {/* <StoryCardLoading /> */}
                </>
            ) : story ? (
                <StoryCardWrapper>{story.title}</StoryCardWrapper>
            ) : null}
        </>
    );
};

// this is going is potentially re-rendered with same props with infinite scroll so keep it in memory
export default memo(HackerNewsStoryCard);
