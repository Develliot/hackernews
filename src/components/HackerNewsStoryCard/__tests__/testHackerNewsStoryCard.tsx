import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';
import HackerNewsStoryCard from '../';

describe('HackerNewsStoryCard', () => {
    const mockProps = {
        isError: false,
        isLoading: false,
        story: {
            by: 'string',
            descendants: 5,
            id: 4,
            kids: [5, 6, 7],
            score: 34,
            time: 344223123,
            title: 'title text',
            type: 'story',
            url: 'http://www.blah.com',
        },
        retry: () => {},
    };

    it('renders correctly when loading', () => {
        const wrapper: ShallowWrapper = shallow(
            <HackerNewsStoryCard {...mockProps} isLoading={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with error', () => {
        const wrapper: ShallowWrapper = shallow(
            <HackerNewsStoryCard {...mockProps} isError={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with story', () => {
        const wrapper: ShallowWrapper = shallow(
            <HackerNewsStoryCard {...mockProps} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
