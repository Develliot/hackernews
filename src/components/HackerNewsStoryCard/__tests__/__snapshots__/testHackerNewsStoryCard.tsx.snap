// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HackerNewsStoryCard renders correctly when loading 1`] = `
<Fragment>
  <span>
    Loading story
  </span>
</Fragment>
`;

exports[`HackerNewsStoryCard renders correctly with error 1`] = `
<Fragment>
  <ErrorMessage
    errorMessage="Unable to retrieve story"
    retry={[Function]}
  />
</Fragment>
`;

exports[`HackerNewsStoryCard renders correctly with story 1`] = `
<Fragment>
  <styled.article
    onClick={[Function]}
    selected={false}
  >
    <styled.div>
      <Typography__Paragraph
        color="white"
        textAlign="center"
      >
        Score
      </Typography__Paragraph>
      <Typography__Paragraph
        color="white"
        textAlign="center"
      >
        34
      </Typography__Paragraph>
    </styled.div>
    <styled.div>
      <styled.a
        href="http://www.blah.com"
        target="_blank"
      >
        <Styled(Typography__H4)
          color="white"
        >
          title text
        </Styled(Typography__H4)>
      </styled.a>
      <styled.div>
        <Styled(Typography__Paragraph)
          color="white"
        >
          <Typography__Span
            color="white"
          >
            By 
          </Typography__Span>
          <Styled(Typography__Span)
            bold={true}
            color="white"
          >
            string
          </Styled(Typography__Span)>
        </Styled(Typography__Paragraph)>
      </styled.div>
    </styled.div>
  </styled.article>
</Fragment>
`;
